internal/session/repository_connection.go

```diff
@@ -48,7 +49,13 @@ func (r *Repository) ListConnections(ctx context.Context, sessionId string, opt
 	const op = "session.(Repository).ListConnections"
 	opts := getOpts(opt...)
 	var connections []*Connection
-	err := r.list(ctx, &connections, "session_id = ?", []interface{}{sessionId}, opts) // pass options, so WithLimit and WithOrder are supported
+	var args []interface{}
+	if sessionId != "" {
+		args = append(args, sessionId)
+	}
+	// The where clause will be ignored if args is empty, if we don't want to
```

Comment:
This essentially makes the `sessionId` parameter optional and allows all of the rows in the `session_connection` table to be returned. Why do we want to do that?

Reply:
The behavior of the where clause being ignored if args are empty is _already_ how it behaves. The existing code doesn't check whether session ID is empty or not. What my change does is make it so that if you pass in an empty session ID, instead of getting nothing back (because it puts an empty string in as the session ID, you get all connections back. In my mind, this matches the comment at the top -- this function lists connections. Session ID therefore is a constraint.

In support of that point, it will be useful at some point in the near future to be able to filter by server ID instead of by session ID. That puts us in the realm of being able to specify one constraint or the other or both.

Reply:
The current comment at the top is "ListConnections will sessions.  Supports the WithLimit and WithOrder options." which doesn't make any sense let alone describe the expected behavior. In general, I think we use Option for optional parameters and treat the other parameters as required. Your change makes `sessionId` an optional parameter and changes the meaning of the method from "lookup connections for session id" to "get all connections".

Reply:
As discussed on Slack I actually think it _should_ be optional and the current behavior (and API signature) is incorrect. However for now I've changed it so that passing `*` explicitly lists all connections, rather than passing no value.

Reply:
I think we should split this out to a new method instead of using a `*`.

Reply:
Meaning, we should create a `GetAllConnections` method and if it is only needed for testing, add it should be in a `_test.go` file and unexported.

Reply:
Didn't see this comment at the time of merge; addressed by https://github.com/hashicorp/boundary/pull/1224